implementation.js

```diff
@@ -73,12 +75,12 @@ if (hasSymbols) {
 	if (supportsStrIterator) {
 		iteratorSymbol = '@@iterator';
 	} else if ('Set' in global && 'Map' in global) {
-		var map = new Map();
-		map.set(1, 2);
-		var y;
-		Function('map', 'y', 'for (var x of map) {y = x}')(map, y); // eslint-disable-line no-new-func
-		if (y[0] === 1 && y[1] === 2) {
-			forOfOnly = true;
+		try {
+			var x = Function('for (var x of new Map().set(1, 2)) return x;')(); // eslint-disable-line no-new-func
```

Comment:
i'd prefer to put as little code in the eval as possible, ie, pass the map as an argument. In addition, it'd be nice to not rely on `Map#set` returning the instance, since I'm not sure that happens in every browser.